snapcraft/plugins/v1/catkin.py

```diff
@@ -321,6 +334,10 @@ def __init__(self, name, options, project):
         self.catkin_packages = None
         if options.catkin_packages is not None:
             self.catkin_packages = set(options.catkin_packages)
+
+        # Get a unique set of skip-keys
+        self.skip_keys = set(options.skip_keys)
```

Comment:
Ah, you've removed too much, you still have the same bug now, but in a different place: if `options.skip_keys` is `None`, this will raise. This needs to be:

```python
self.skip_keys = set()
if options.skip_keys is not None:
    self.skip_keys = set(options.skip_keys)
```